### PR TITLE
ARQ-2114 It was easy to reach the GH rate limit when firefox is used in large test suites running in CI

### PR DESCRIPTION
**Pull request by MatousJobanek**
- When a request with the first page is sent, then the parameter is not added -the reason is that conditional requests don't seem to be supported for parametrized requests

- When GH release is retrieved for a version then it check for the presence every page separately (lazily)

- When version is used, check for previous downloads is performed before getting the download URL

- Added logging about sending requests

### CODE REVIEW

File: drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/utils/HttpClient.java [hunk 1]
```diff
@@ -22,6 +27,7 @@ public Response get(String url, Map<String, String> headers) throws IOException
         try (CloseableHttpClient client = createHttpClient()) {
             final HttpGet request = new HttpGet(url);
             addHeaders(headers, request);
+            log.info("Sending request: " + request + " with headers: " + Arrays.asList(request.getAllHeaders()));
```
Comment (bartoszmajsak) [inline drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/utils/HttpClient.java]: Do we really need to make it `info` level?

**Comment by bartoszmajsak** [comment]
Can we also think about letting users refer to their GH tokens to be used? This would solve a problem of any other potential GH calls we will use in the future.

File: drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/utils/HttpClient.java (hunk 1 above)
Comment (MatousJobanek) [inline drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/utils/HttpClient.java]: I'm not sure about this. WDYT? which level would be more appropriate? Debug? Fine?

**Comment by MatousJobanek** [comment]
Yeah, it would solve the problem. I was always trying to avoid it - not to use the credentials in CI.

File: drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/utils/HttpClient.java (hunk 1 above)
Comment (bartoszmajsak) [inline drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/utils/HttpClient.java]: Personally I am not interested to see if requests are sent. Only when something is not working I would be interested in seeing this information - so maybe debug with a simple way to enable it?

**Comment by MatousJobanek** [comment]
In this case, the main problem was when some specific version was used - I forgot to check the directory as the first thing - so it led to the state when two requests were sent in every run.

File: drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/utils/HttpClient.java (hunk 1 above)
Comment (MatousJobanek) [inline drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/utils/HttpClient.java]: > simple way to enable it?

:-D this is the reason why I didn't want to use a lower level... but I could use debug level or INFO when -Darquillian.debug is used

**Pull request merged**